File: java/src/main/java/money_problem/domain/MoneyCalculator.java
package money_problem.domain;

public class MoneyCalculator {
    public static double add(double amount, Currency currency, double amount2) {
        return amount + amount2;
    }

    public static double times(double amount, Currency currency, int value) {
        return amount * value;
    }

    public static double divide(double amount, Currency currency, int value) {
        return amount / value;
    }
}

File: csharp/money-problem.Domain/MoneyCalculator.cs
namespace money_problem.Domain;

public static class MoneyCalculator
{
    public static double Add(double amount, Currency currency, double amount2) => amount + amount2;
    public static double Times(double amount, Currency currency, int value) => amount * value;
    public static double Divide(double amount, Currency currency, int value) => amount / value;
}
